protos/feast/core/Permission.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+package feast.core;
+
+option go_package = "github.com/feast-dev/feast/go/protos/feast/core";
+option java_outer_classname = "Permission";
+option java_package = "feast.proto.core";
+
+import "feast/core/Policy.proto";
+
+message Permission {
+
+  enum AuthzedAction {
+    CREATE = 0;
+    READ = 1;
+    UPDATE = 2;
+    DELETE = 3;
+    QUERY_ONLINE = 4;
+    QUERY_OFFLINE = 5;
+    WRITE_ONLINE = 6;
+    WRITE_OFFLINE = 7;
+  }
+
+  // Name of the permission. Must be unique. Not updated.
+  string name = 1;
+
+  // Name of Feast project.
+  string project = 2;
+
+  enum Type {
+      FEATURE_VIEW = 0;
+      ON_DEMAND_FEATURE_VIEW = 1;
+      BATCH_FEATURE_VIEW = 2;
+      STREAM_FEATURE_VIEW= 3;
+      ENTITY = 4;
+      FEATURE_SERVICE = 5;
+      DATA_SOURCE = 6;
+      VALIDATION_REFERENCE = 7;
+      SAVED_DATASET = 8;
+      PERMISSION = 9;
+  }
+
+  repeated Type types = 3;
+
+  bool with_subclasses = 4;
```

Comment:
I like the idea of allowing all fv types to be configured together, but tying it to subclasses is gonna bite us in the end. odfvs don't extend `FeatureView` for example, while `StreamFeatureView` does. I think we shouldn't rely on that.

Reply:
Do you mean to use only `FeatureView` type for all the possible fvs, whether they extend `FeatureView` or not?
And also provide a more punctual option for extensions like odfv and similar, or expose only the `FeatureView` type?
It'd be great if you can clarify your ask so we can adapt the code properly.

Reply:
I'm leaning towards removing `with_subclasses` and instead adding some sort of `ALL_FEATURE_VIEWS` alias/helper, similar to `WRITE` and `QUERY`. wdyt?

Reply:
IMO, if we remove `with_subclasses`, the logic should be as if the flag was set to `True` (`OnDemandFeatureView` would be excluded by `type=FeatureView`, as you said before).
I would prefer a simple, understandable, filtering logic that applies to all the resource types, with no exceptions: consider that `DataSource` is just an abstract class and we don't have the actual classes in `FeastObject` type. To match data sources types we need to include the subclasses, which explains the reasons for my personal preference.

Reply:
Didn't think about DataSources.. yeah, that complicates things. Maybe the solution could be to rewrite `StreamFeatureView` to extend `BaseFeatureView` instead of `FeatureView`. That feels like how it should have been done from the get-go. Let me look into that and circle back.

Reply:
I opened #4398 to track this. I think we can for now continue as if that's already the case and we don't have inconsistent class inheritance. In other words, let's remove `with_subclasses` and checks will naturally default to True as you suggested.

Reply:
I will also add a global variable like ALL_FEATURE_VIEWS to explicitely list all of them.
Thanks

Reply:
see [latest commit](https://github.com/RHEcosystemAppEng/feast/pull/74/commits/0033c1fe39e4a9ba26c26c7d4c64cef95d14f981)

Reply:
great, just a single nit. Probably best to remove this sentence altogether `Please note that all sub-classes of FeatureView are also included.` since `FeatureView` will no longer have any subclasses after the refactor.